From 3f955f47fa7534e0bf58d7f51ebf6c3ec76c07ca Mon Sep 17 00:00:00 2001
From: UN1CA <unica@unica.org>
Date: Wed, 15 Jul 2026 19:40:00 +0200
Subject: [PATCH] Correct Now Nudge conversation resolver pipeline

Keep messaging applications on the ContentCapture conversation path instead
of fabricating generic autofill input types. Restore semantic recall, preserve
the normal SmartReply state gates, initialize those gates for parser-created
chat sessions, and dispatch each parsed conversation once.
---

diff --git a/smali_classes15/com/samsung/android/smartsuggestions/service/autofill/AutofillNudgeManager.smali b/smali_classes15/com/samsung/android/smartsuggestions/service/autofill/AutofillNudgeManager.smali
index 0bb3164..753840b 100644
--- a/smali_classes15/com/samsung/android/smartsuggestions/service/autofill/AutofillNudgeManager.smali
+++ b/smali_classes15/com/samsung/android/smartsuggestions/service/autofill/AutofillNudgeManager.smali
@@ -348,6 +348,73 @@
 
     invoke-static {v3, p1}, Lkotlin/jvm/internal/h;->p(Ljava/lang/Object;Ljava/lang/String;)V
 
+    const-string v0, "com.samsung.android.messaging"
+
+    invoke-static {v3, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
+
+    move-result v0
+
+    if-nez v0, :cond_rezoss_skip_chat_autofill
+
+    const-string v0, "com.whatsapp"
+
+    invoke-static {v3, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
+
+    move-result v0
+
+    if-nez v0, :cond_rezoss_skip_chat_autofill
+
+    const-string v0, "com.whatsapp.w4b"
+
+    invoke-static {v3, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
+
+    move-result v0
+
+    if-nez v0, :cond_rezoss_skip_chat_autofill
+
+    const-string v0, "org.telegram.messenger"
+
+    invoke-static {v3, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
+
+    move-result v0
+
+    if-nez v0, :cond_rezoss_skip_chat_autofill
+
+    const-string v0, "org.telegram.messenger.web"
+
+    invoke-static {v3, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
+
+    move-result v0
+
+    if-nez v0, :cond_rezoss_skip_chat_autofill
+
+    const-string v0, "org.telegram.plus"
+
+    invoke-static {v3, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
+
+    move-result v0
+
+    if-nez v0, :cond_rezoss_skip_chat_autofill
+
+    const-string v0, "org.thunderdog.challegram"
+
+    invoke-static {v3, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
+
+    move-result v0
+
+    if-eqz v0, :cond_rezoss_continue_autofill
+
+    :cond_rezoss_skip_chat_autofill
+    const-string p1, "AutofillNudgeManager"
+
+    const-string p2, "Skip generic autofill for ContentCapture SmartReply package"
+
+    invoke-static {p1, p2}, Lcom/samsung/android/smartsuggestions/service/autofill/util/AutofillFlowLogger;->i(Ljava/lang/String;Ljava/lang/String;)V
+
+    return-void
+
+    :cond_rezoss_continue_autofill
+
     sget-object p1, Lcom/samsung/android/smartsuggestions/service/policy/AutofillPolicyManager;->Companion:Lcom/samsung/android/smartsuggestions/service/policy/AutofillPolicyManager$Companion;
 
     iget-object p2, p0, Lcom/samsung/android/smartsuggestions/service/autofill/AutofillNudgeManager;->context:Landroid/content/Context;
@@ -515,9 +582,11 @@
 
     invoke-static {v4, p1}, Lcom/samsung/android/smartsuggestions/service/autofill/util/AutofillFlowLogger;->e(Ljava/lang/String;Ljava/lang/String;)V
 
-    const-string p0, "Missing autofill node; keep Now Nudge flow alive"
+    sget-object p1, Lcom/samsung/android/smartsuggestions/service/autofill/util/AutofillNudgeHelper;->INSTANCE:Lcom/samsung/android/smartsuggestions/service/autofill/util/AutofillNudgeHelper;
+
+    iget-object p0, p0, Lcom/samsung/android/smartsuggestions/service/autofill/AutofillNudgeManager;->context:Landroid/content/Context;
 
-    invoke-static {v4, p0}, Lcom/samsung/android/smartsuggestions/service/autofill/util/AutofillFlowLogger;->i(Ljava/lang/String;Ljava/lang/String;)V
+    invoke-virtual {p1, p0}, Lcom/samsung/android/smartsuggestions/service/autofill/util/AutofillNudgeHelper;->hideInlineSuggestion(Landroid/content/Context;)V
 
     :cond_3
     return-void
diff --git a/smali_classes15/com/samsung/android/smartsuggestions/service/autofill/collector/AutofillEventController.smali b/smali_classes15/com/samsung/android/smartsuggestions/service/autofill/collector/AutofillEventController.smali
index 5831d73..1e228d5 100644
--- a/smali_classes15/com/samsung/android/smartsuggestions/service/autofill/collector/AutofillEventController.smali
+++ b/smali_classes15/com/samsung/android/smartsuggestions/service/autofill/collector/AutofillEventController.smali
@@ -782,94 +782,6 @@
     return p0
 .end method
 
-.method private final isWhatsappEntry(Lcom/samsung/android/smartsuggestions/service/autofill/data/AssistData;)Z
-    .locals 4
-
-    invoke-virtual {p1}, Lcom/samsung/android/smartsuggestions/service/autofill/data/AssistData;->getComponentName()Landroid/content/ComponentName;
-
-    move-result-object p0
-
-    invoke-virtual {p0}, Landroid/content/ComponentName;->getPackageName()Ljava/lang/String;
-
-    move-result-object p0
-
-    const-string v0, "com.whatsapp"
-
-    invoke-static {p0, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_rezoss_package_ok
-
-    const-string v0, "com.whatsapp.w4b"
-
-    invoke-static {p0, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result p0
-
-    if-eqz p0, :cond_rezoss_false
-
-    :cond_rezoss_package_ok
-    invoke-virtual {p1}, Lcom/samsung/android/smartsuggestions/service/autofill/data/AssistData;->getNodeInfo()Landroid/app/assist/AssistStructure$ViewNode;
-
-    move-result-object p0
-
-    invoke-virtual {p0}, Landroid/app/assist/AssistStructure$ViewNode;->getAutofillType()I
-
-    move-result p1
-
-    const/4 v2, 0x1
-
-    if-ne p1, v2, :cond_rezoss_false
-
-    invoke-virtual {p0}, Landroid/app/assist/AssistStructure$ViewNode;->getIdEntry()Ljava/lang/String;
-
-    move-result-object p1
-
-    const-string v3, "entry"
-
-    invoke-static {p1, v3}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v0
-
-    if-nez v0, :cond_rezoss_true
-
-    if-eqz p1, :cond_rezoss_check_text_id
-
-    invoke-static {p1, v3}, Lkotlin/text/v;->l0(Ljava/lang/String;Ljava/lang/String;)Z
-
-    move-result p1
-
-    if-nez p1, :cond_rezoss_true
-
-    :cond_rezoss_check_text_id
-    invoke-virtual {p0}, Landroid/app/assist/AssistStructure$ViewNode;->getTextIdEntry()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {p0, v3}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result p1
-
-    if-nez p1, :cond_rezoss_true
-
-    if-eqz p0, :cond_rezoss_false
-
-    invoke-static {p0, v3}, Lkotlin/text/v;->l0(Ljava/lang/String;Ljava/lang/String;)Z
-
-    move-result p0
-
-    if-nez p0, :cond_rezoss_true
-
-    :cond_rezoss_false
-    const/4 p0, 0x0
-
-    return p0
-
-    :cond_rezoss_true
-    return v2
-.end method
-
 .method private final matched(Landroid/view/autofill/AutofillId;Ljava/util/Set;)Z
     .locals 3
     .annotation system Ldalvik/annotation/Signature;
@@ -1008,33 +920,6 @@
     return-void
 
     :cond_0
-    invoke-direct {p0, p1}, Lcom/samsung/android/smartsuggestions/service/autofill/collector/AutofillEventController;->isWhatsappEntry(Lcom/samsung/android/smartsuggestions/service/autofill/data/AssistData;)Z
-
-    move-result v0
-
-    if-eqz v0, :cond_rezoss_hide_inline
-
-    new-instance v0, Ljava/util/LinkedHashMap;
-
-    invoke-direct {v0}, Ljava/util/LinkedHashMap;-><init>()V
-
-    sget-object v1, Lcom/samsung/android/smartsuggestions/service/autofill/data/FieldInputType;->UNKNOWN:Lcom/samsung/android/smartsuggestions/service/autofill/data/FieldInputType;
-
-    const-string v2, "message"
-
-    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
-
-    const-string v1, "AutofillEventController"
-
-    const-string v2, "WhatsApp entry TEXT has no PDE userContext; synthesize chat/message context"
-
-    invoke-static {v1, v2}, Lcom/samsung/android/smartsuggestions/service/autofill/util/AutofillFlowLogger;->i(Ljava/lang/String;Ljava/lang/String;)V
-
-    invoke-direct {p0, p1, v0}, Lcom/samsung/android/smartsuggestions/service/autofill/collector/AutofillEventController;->showNudge(Lcom/samsung/android/smartsuggestions/service/autofill/data/AssistData;Ljava/util/Map;)V
-
-    return-void
-
-    :cond_rezoss_hide_inline
     sget-object p1, Lcom/samsung/android/smartsuggestions/service/autofill/util/AutofillNudgeHelper;->INSTANCE:Lcom/samsung/android/smartsuggestions/service/autofill/util/AutofillNudgeHelper;
 
     iget-object p0, p0, Lcom/samsung/android/smartsuggestions/service/autofill/collector/AutofillEventController;->context:Landroid/content/Context;
diff --git a/smali_classes15/com/samsung/android/smartsuggestions/service/autofill/interpreter/AssistDataInterpreter.smali b/smali_classes15/com/samsung/android/smartsuggestions/service/autofill/interpreter/AssistDataInterpreter.smali
index 667ff2e..7028b05 100644
--- a/smali_classes15/com/samsung/android/smartsuggestions/service/autofill/interpreter/AssistDataInterpreter.smali
+++ b/smali_classes15/com/samsung/android/smartsuggestions/service/autofill/interpreter/AssistDataInterpreter.smali
@@ -650,230 +650,6 @@
     return p0
 .end method
 
-.method private final shouldSkipUnsupportedChatFieldHide(Lcom/samsung/android/smartsuggestions/service/autofill/data/AssistData;)Z
-    .locals 2
-
-    invoke-virtual {p1}, Lcom/samsung/android/smartsuggestions/service/autofill/data/AssistData;->getNodeInfo()Landroid/app/assist/AssistStructure$ViewNode;
-
-    move-result-object p0
-
-    invoke-virtual {p0}, Landroid/app/assist/AssistStructure$ViewNode;->getAutofillType()I
-
-    move-result p0
-
-    const/4 v0, 0x1
-
-    if-ne p0, v0, :cond_not_chat_package
-
-    invoke-virtual {p1}, Lcom/samsung/android/smartsuggestions/service/autofill/data/AssistData;->getComponentName()Landroid/content/ComponentName;
-
-    move-result-object p0
-
-    invoke-virtual {p0}, Landroid/content/ComponentName;->getPackageName()Ljava/lang/String;
-
-    move-result-object p0
-
-    const-string p1, "com.whatsapp"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.whatsapp.w4b"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.google.android.apps.messaging"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.google.android.apps.dynamite"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.kakao.talk"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "jp.naver.line.android"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "org.thoughtcrime.securesms"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.instagram.android"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.sgiggle.production"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.nttdocomo.android.msg"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.kddi.android.cmail"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.verizon.messaging.vzmsgs"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "org.telegram.messenger"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "org.telegram.messenger.web"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.samsung.android.messaging"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.facebook.orca"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.tencent.mm"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.viber.voip"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.discord"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.snapchat.android"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.microsoft.teams"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "com.google.android.apps.googlevoice"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "org.telegram.plus"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_chat_package
-
-    const-string p1, "org.thunderdog.challegram"
-
-    invoke-static {p0, p1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result p0
-
-    if-nez p0, :cond_chat_package
-
-    :cond_not_chat_package
-    const/4 p0, 0x0
-
-    return p0
-
-    :cond_chat_package
-    return v0
-.end method
-
 .method private final isFocusedFieldSupported(Lcom/samsung/android/smartsuggestions/service/autofill/data/AssistData;)Z
     .locals 1
 
@@ -1528,22 +1304,6 @@
 
     if-nez v0, :cond_0
 
-    invoke-direct {p0, p1}, Lcom/samsung/android/smartsuggestions/service/autofill/interpreter/AssistDataInterpreter;->shouldSkipUnsupportedChatFieldHide(Lcom/samsung/android/smartsuggestions/service/autofill/data/AssistData;)Z
-
-    move-result v0
-
-    if-eqz v0, :cond_hide_unsupported_field
-
-    const-string v0, "Unsupported chat field; continue Now Nudge flow"
-
-    invoke-static {v1, v0}, Lcom/samsung/android/smartsuggestions/service/autofill/util/AutofillFlowLogger;->i(Ljava/lang/String;Ljava/lang/String;)V
-
-    const/4 v0, 0x1
-
-    goto :cond_0
-
-    :cond_hide_unsupported_field
-
     const-string p1, "Focused field is not supported"
 
     invoke-static {v1, p1}, Lcom/samsung/android/smartsuggestions/service/autofill/util/AutofillFlowLogger;->i(Ljava/lang/String;Ljava/lang/String;)V
diff --git a/smali_classes15/com/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/ScreenContentManager.smali b/smali_classes15/com/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/ScreenContentManager.smali
index 742e38c..b8a8440 100644
--- a/smali_classes15/com/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/ScreenContentManager.smali
+++ b/smali_classes15/com/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/ScreenContentManager.smali
@@ -1982,68 +1982,6 @@
 
     if-nez v0, :cond_0
 
-    invoke-virtual {p3}, Landroid/content/ComponentName;->getPackageName()Ljava/lang/String;
-
-    move-result-object v0
-
-    const-string v1, "com.whatsapp"
-
-    invoke-static {v0, v1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_rezoss_direct_whatsapp_smartreply
-
-    const-string v1, "org.telegram.messenger"
-
-    invoke-static {v0, v1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_rezoss_direct_whatsapp_smartreply
-
-    const-string v1, "org.telegram.messenger.web"
-
-    invoke-static {v0, v1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-nez v1, :cond_rezoss_direct_whatsapp_smartreply
-
-    const-string v1, "com.whatsapp.w4b"
-
-    invoke-static {v0, v1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v0
-
-    if-eqz v0, :cond_rezoss_direct_whatsapp_done
-
-    :cond_rezoss_direct_whatsapp_smartreply
-    sget-object v2, Lcom/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/SmartReplyHandler;->INSTANCE:Lcom/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/SmartReplyHandler;
-
-    iget-object v0, p0, Lcom/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/ScreenContentManager;->chatTranslationObserver:Ljava/util/concurrent/CopyOnWriteArrayList;
-
-    invoke-virtual {v0, v2}, Ljava/util/concurrent/CopyOnWriteArrayList;->remove(Ljava/lang/Object;)Z
-
-    iget-object v4, p0, Lcom/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/ScreenContentManager;->context:Landroid/content/Context;
-
-    move-object v3, p1
-
-    move-object v5, p2
-
-    move-object v6, p3
-
-    move-object v7, p4
-
-    invoke-interface/range {v2 .. v7}, Lcom/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/ChatTranslationObserver;->onChatTranslationMessages(Landroid/view/contentcapture/ContentCaptureSessionId;Landroid/content/Context;Ljava/util/List;Landroid/content/ComponentName;Landroid/graphics/Rect;)V
-
-    const-string v0, "ScreenContentManager"
-
-    const-string/jumbo v1, "direct chat SmartReply dispatch completed"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    :cond_rezoss_direct_whatsapp_done
     iget-object v0, p0, Lcom/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/ScreenContentManager;->chatTranslationObserver:Ljava/util/concurrent/CopyOnWriteArrayList;
 
     invoke-interface {v0}, Ljava/lang/Iterable;->iterator()Ljava/util/Iterator;
diff --git a/smali_classes15/com/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/SmartReplyHandler.smali b/smali_classes15/com/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/SmartReplyHandler.smali
index e53351d..0d53151 100644
--- a/smali_classes15/com/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/SmartReplyHandler.smali
+++ b/smali_classes15/com/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/SmartReplyHandler.smali
@@ -1870,79 +1870,6 @@
 
     move-object/from16 v2, p3
 
-    invoke-static/range {p1 .. p1}, Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;->getInstance(Landroid/content/Context;)Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;
-
-    move-result-object v5
-
-    invoke-virtual {v5}, Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;->getPolicyListSync()Lcom/samsung/android/smartsuggestions/service/policy/Policylist;
-
-    move-result-object v5
-
-    if-eqz v5, :cond_rezoss_start_policy_done
-
-    invoke-virtual {v5}, Lcom/samsung/android/smartsuggestions/service/policy/Policylist;->getSuggestedReplyPackageAllowlist()Ljava/util/List;
-
-    move-result-object v5
-
-    invoke-interface {v5, v2}, Ljava/util/List;->contains(Ljava/lang/Object;)Z
-
-    move-result v5
-
-    if-eqz v5, :cond_rezoss_start_policy_done
-
-    goto :cond_rezoss_start_force_whatsapp
-
-    :cond_rezoss_start_policy_done
-    const-string v5, "com.whatsapp"
-
-    invoke-static {v2, v5}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v5
-
-    if-nez v5, :cond_rezoss_start_force_whatsapp
-
-    const-string v5, "org.telegram.messenger"
-
-    invoke-static {v2, v5}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v5
-
-    if-nez v5, :cond_rezoss_start_force_whatsapp
-
-    const-string v5, "org.telegram.messenger.web"
-
-    invoke-static {v2, v5}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v5
-
-    if-nez v5, :cond_rezoss_start_force_whatsapp
-
-    const-string v5, "com.whatsapp.w4b"
-
-    invoke-static {v2, v5}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v5
-
-    if-eqz v5, :cond_rezoss_start_original_enabled
-
-    :cond_rezoss_start_force_whatsapp
-    const/4 v1, 0x1
-
-    sput-boolean v1, Lcom/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/SmartReplyHandler;->isNowNudgeOn:Z
-
-    sput-boolean v1, Lcom/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/SmartReplyHandler;->isKeyboardVisible:Z
-
-    const-string v5, "SmartReplyHandler[reply]"
-
-    const-string/jumbo v6, "force-enabled chat startSmartReply gate"
-
-    invoke-static {v5, v6}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    const/4 v4, 0x0
-
-    goto :goto_0
-
-    :cond_rezoss_start_original_enabled
     invoke-virtual {v1, v2}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
 
     move-result-object v1
@@ -2450,6 +2377,12 @@
 
     invoke-virtual {v0, p1, p4}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
 
+    invoke-virtual {p0, p2}, Lcom/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/SmartReplyHandler;->updateSmartReplyOnOff(Landroid/content/Context;)V
+
+    invoke-virtual {p0, p2}, Lcom/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/SmartReplyHandler;->updateNowNudgeOnOff(Landroid/content/Context;)V
+
+    invoke-direct {p0, p2, p4}, Lcom/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/SmartReplyHandler;->updateEnableList(Landroid/content/Context;Landroid/content/ComponentName;)V
+
     sget-object v0, Lcom/samsung/android/smartsuggestions/service/deepscreencapture/screenunderstanding/SmartReplyHandler;->newMessagesSessionMap:Ljava/util/HashMap;
 
     invoke-virtual {v0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
diff --git a/smali_classes15/com/samsung/android/smartsuggestions/service/inlinesuggestions/data/AssistStructureFinder.smali b/smali_classes15/com/samsung/android/smartsuggestions/service/inlinesuggestions/data/AssistStructureFinder.smali
index aa32882..1f5e63e 100644
--- a/smali_classes15/com/samsung/android/smartsuggestions/service/inlinesuggestions/data/AssistStructureFinder.smali
+++ b/smali_classes15/com/samsung/android/smartsuggestions/service/inlinesuggestions/data/AssistStructureFinder.smali
@@ -82,136 +82,6 @@
     return-void
 .end method
 
-.method private final findWhatsappComposeNode(Landroid/app/assist/AssistStructure$ViewNode;)Landroid/app/assist/AssistStructure$ViewNode;
-    .locals 4
-
-    if-eqz p1, :cond_rezoss_not_found
-
-    invoke-direct {p0, p1}, Lcom/samsung/android/smartsuggestions/service/inlinesuggestions/data/AssistStructureFinder;->isWhatsappComposeNode(Landroid/app/assist/AssistStructure$ViewNode;)Z
-
-    move-result v0
-
-    if-eqz v0, :cond_rezoss_check_children
-
-    const-string v0, "AssistStructureFinder"
-
-    const-string v1, "Recovered WhatsApp compose node for collectAssistData"
-
-    invoke-static {v0, v1}, Lxc/e;->c(Ljava/lang/String;Ljava/lang/String;)V
-
-    invoke-direct {p0, p1}, Lcom/samsung/android/smartsuggestions/service/inlinesuggestions/data/AssistStructureFinder;->printViewNode(Landroid/app/assist/AssistStructure$ViewNode;)V
-
-    return-object p1
-
-    :cond_rezoss_check_children
-    invoke-virtual {p1}, Landroid/app/assist/AssistStructure$ViewNode;->getChildCount()I
-
-    move-result v0
-
-    const/4 v1, 0x0
-
-    :goto_rezoss_child_loop
-    if-ge v1, v0, :cond_rezoss_not_found
-
-    invoke-virtual {p1, v1}, Landroid/app/assist/AssistStructure$ViewNode;->getChildAt(I)Landroid/app/assist/AssistStructure$ViewNode;
-
-    move-result-object v2
-
-    invoke-direct {p0, v2}, Lcom/samsung/android/smartsuggestions/service/inlinesuggestions/data/AssistStructureFinder;->findWhatsappComposeNode(Landroid/app/assist/AssistStructure$ViewNode;)Landroid/app/assist/AssistStructure$ViewNode;
-
-    move-result-object v3
-
-    if-eqz v3, :cond_rezoss_next_child
-
-    return-object v3
-
-    :cond_rezoss_next_child
-    add-int/lit8 v1, v1, 0x1
-
-    goto :goto_rezoss_child_loop
-
-    :cond_rezoss_not_found
-    const/4 p0, 0x0
-
-    return-object p0
-.end method
-
-.method private final isWhatsappComposeNode(Landroid/app/assist/AssistStructure$ViewNode;)Z
-    .locals 3
-
-    invoke-virtual {p1}, Landroid/app/assist/AssistStructure$ViewNode;->getIdEntry()Ljava/lang/String;
-
-    move-result-object v0
-
-    const-string v1, "entry"
-
-    invoke-static {v0, v1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v2
-
-    if-nez v2, :cond_rezoss_true
-
-    if-eqz v0, :cond_rezoss_check_text_id
-
-    invoke-static {v0, v1}, Lkotlin/text/v;->l0(Ljava/lang/String;Ljava/lang/String;)Z
-
-    move-result v2
-
-    if-nez v2, :cond_rezoss_check_class
-
-    :cond_rezoss_check_text_id
-    invoke-virtual {p1}, Landroid/app/assist/AssistStructure$ViewNode;->getTextIdEntry()Ljava/lang/String;
-
-    move-result-object v0
-
-    invoke-static {v0, v1}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v2
-
-    if-nez v2, :cond_rezoss_true
-
-    if-eqz v0, :cond_rezoss_check_class
-
-    invoke-static {v0, v1}, Lkotlin/text/v;->l0(Ljava/lang/String;Ljava/lang/String;)Z
-
-    move-result v2
-
-    if-nez v2, :cond_rezoss_check_class
-
-    :cond_rezoss_check_class
-    invoke-virtual {p1}, Landroid/app/assist/AssistStructure$ViewNode;->getAutofillType()I
-
-    move-result p0
-
-    const/4 v2, 0x1
-
-    if-ne p0, v2, :cond_rezoss_false
-
-    invoke-virtual {p1}, Landroid/app/assist/AssistStructure$ViewNode;->getClassName()Ljava/lang/String;
-
-    move-result-object v0
-
-    if-eqz v0, :cond_rezoss_false
-
-    const-string v1, "EditText"
-
-    invoke-static {v0, v1}, Lkotlin/text/v;->l0(Ljava/lang/String;Ljava/lang/String;)Z
-
-    move-result v0
-
-    if-nez v0, :cond_rezoss_true
-
-    :cond_rezoss_false
-    const/4 p0, 0x0
-
-    return p0
-
-    :cond_rezoss_true
-    const/4 p0, 0x1
-
-    return p0
-.end method
-
 .method private final findNode(Ljava/lang/String;Landroid/app/assist/AssistStructure$ViewNode;)Landroid/app/assist/AssistStructure$ViewNode;
     .locals 4
 
@@ -359,7 +229,7 @@
 
 # virtual methods
 .method public final find(Landroid/content/ComponentName;Ljava/lang/String;)Landroid/app/assist/AssistStructure$ViewNode;
-    .locals 7
+    .locals 5
     .param p1    # Landroid/content/ComponentName;
         .annotation build Lorg/jetbrains/annotations/NotNull;
         .end annotation
@@ -379,30 +249,6 @@
 
     invoke-static {p2, v0}, Lkotlin/jvm/internal/h;->q(Ljava/lang/Object;Ljava/lang/String;)V
 
-    invoke-virtual {p1}, Landroid/content/ComponentName;->getPackageName()Ljava/lang/String;
-
-    move-result-object v5
-
-    const-string v0, "com.whatsapp"
-
-    invoke-static {v5, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v0
-
-    if-nez v0, :cond_rezoss_whatsapp_package
-
-    const-string v0, "com.whatsapp.w4b"
-
-    invoke-static {v5, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v5
-
-    goto :goto_rezoss_package_checked
-
-    :cond_rezoss_whatsapp_package
-    const/4 v5, 0x1
-
-    :goto_rezoss_package_checked
     invoke-static {}, Lcom/samsung/android/smartsuggestions/service/deepscreencapture/assist/AssistHolder;->getInstance()Lcom/samsung/android/smartsuggestions/service/deepscreencapture/assist/AssistHolder;
 
     move-result-object v0
@@ -468,8 +314,6 @@
 
     invoke-static {v2}, Lkotlin/jvm/internal/h;->n(Ljava/lang/Object;)V
 
-    move-object v6, v2
-
     invoke-direct {p0, p2, v2}, Lcom/samsung/android/smartsuggestions/service/inlinesuggestions/data/AssistStructureFinder;->findNode(Ljava/lang/String;Landroid/app/assist/AssistStructure$ViewNode;)Landroid/app/assist/AssistStructure$ViewNode;
 
     move-result-object v2
@@ -483,21 +327,6 @@
     return-object v2
 
     :cond_1
-    if-eqz v5, :cond_rezoss_next_window
-
-    invoke-direct {p0, v6}, Lcom/samsung/android/smartsuggestions/service/inlinesuggestions/data/AssistStructureFinder;->findWhatsappComposeNode(Landroid/app/assist/AssistStructure$ViewNode;)Landroid/app/assist/AssistStructure$ViewNode;
-
-    move-result-object v2
-
-    if-eqz v2, :cond_rezoss_next_window
-
-    const-string p0, "!!! whatsapp compose fallback find"
-
-    invoke-static {v4, p0}, Lxc/e;->c(Ljava/lang/String;Ljava/lang/String;)V
-
-    return-object v2
-
-    :cond_rezoss_next_window
     add-int/lit8 v1, v1, 0x1
 
     goto :goto_1
diff --git a/smali_classes15/com/samsung/android/smartsuggestions/service/inlinesuggestions/data/ResourceIdEntityMap.smali b/smali_classes15/com/samsung/android/smartsuggestions/service/inlinesuggestions/data/ResourceIdEntityMap.smali
index ff11462..f435553 100644
--- a/smali_classes15/com/samsung/android/smartsuggestions/service/inlinesuggestions/data/ResourceIdEntityMap.smali
+++ b/smali_classes15/com/samsung/android/smartsuggestions/service/inlinesuggestions/data/ResourceIdEntityMap.smali
@@ -137,23 +137,5 @@
     goto :goto_0
 
     :cond_1
-    const-string v1, "com.whatsapp.Conversation_com.whatsapp_entry"
-
-    const-string/jumbo v2, "message"
-
-    invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
-
-    const-string v1, "com.whatsapp.Conversation_com.whatsapp.w4b_entry"
-
-    invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
-
-    const-string v1, "com.whatsapp.w4b.Conversation_com.whatsapp.w4b_entry"
-
-    invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
-
-    const-string v1, "com.whatsapp.w4b.Conversation_com.whatsapp_entry"
-
-    invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
-
     return-object v0
 .end method
diff --git a/smali_classes15/com/samsung/android/smartsuggestions/service/smartreply/SmartReplyRecallHandler.smali b/smali_classes15/com/samsung/android/smartsuggestions/service/smartreply/SmartReplyRecallHandler.smali
index bfa6896..197fa70 100644
--- a/smali_classes15/com/samsung/android/smartsuggestions/service/smartreply/SmartReplyRecallHandler.smali
+++ b/smali_classes15/com/samsung/android/smartsuggestions/service/smartreply/SmartReplyRecallHandler.smali
@@ -3074,81 +3074,6 @@
 
     const-string v5, "context"
 
-    invoke-static {v2}, Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;->getInstance(Landroid/content/Context;)Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;
-
-    move-result-object v5
-
-    invoke-virtual {v5}, Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;->getPolicyListSync()Lcom/samsung/android/smartsuggestions/service/policy/Policylist;
-
-    move-result-object v5
-
-    if-eqz v5, :cond_rezoss_recall_policy_done
-
-    invoke-virtual {v5}, Lcom/samsung/android/smartsuggestions/service/policy/Policylist;->getSuggestedReplyPackageAllowlist()Ljava/util/List;
-
-    move-result-object v5
-
-    invoke-interface {v5, v4}, Ljava/util/List;->contains(Ljava/lang/Object;)Z
-
-    move-result v5
-
-    if-nez v5, :cond_rezoss_skip_chat_recall
-
-    :cond_rezoss_recall_policy_done
-    const-string v5, "com.whatsapp"
-
-    invoke-static {v4, v5}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v5
-
-    if-nez v5, :cond_rezoss_skip_chat_recall
-
-    const-string v5, "org.telegram.messenger"
-
-    invoke-static {v4, v5}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v5
-
-    if-nez v5, :cond_rezoss_skip_chat_recall
-
-    const-string v5, "org.telegram.messenger.web"
-
-    invoke-static {v4, v5}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v5
-
-    if-nez v5, :cond_rezoss_skip_chat_recall
-
-    const-string v5, "com.whatsapp.w4b"
-
-    invoke-static {v4, v5}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v5
-
-    if-nez v5, :cond_rezoss_skip_chat_recall
-
-    const-string v5, "com.samsung.android.messaging"
-
-    invoke-static {v4, v5}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v5
-
-    if-eqz v5, :cond_rezoss_chat_recall_original
-
-    :cond_rezoss_skip_chat_recall
-    const-string v5, "SmartReplyRecallHandler[recall]"
-
-    const-string/jumbo v6, "skip recall/action for chat SmartReply package"
-
-    invoke-static {v5, v6}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    sget-object v5, Lcom/samsung/android/smartsuggestions/service/smartreply/SmartReplyRecallHandler;->EMPTY_RESULT:Lcom/samsung/android/smartsuggestions/service/smartreply/SmartReplyRecallHandler$RecallResults;
-
-    return-object v5
-
-    :cond_rezoss_chat_recall_original
-    const-string v5, "context"
-
     invoke-static {v2, v5}, Lkotlin/jvm/internal/h;->q(Ljava/lang/Object;Ljava/lang/String;)V
 
     const-string v5, "conversation"
diff --git a/smali_classes15/com/samsung/android/smartsuggestions/service/smartreply/SmartReplyResolver.smali b/smali_classes15/com/samsung/android/smartsuggestions/service/smartreply/SmartReplyResolver.smali
index 19aeb03..5ec6209 100644
--- a/smali_classes15/com/samsung/android/smartsuggestions/service/smartreply/SmartReplyResolver.smali
+++ b/smali_classes15/com/samsung/android/smartsuggestions/service/smartreply/SmartReplyResolver.smali
@@ -1837,79 +1837,6 @@
     :goto_c
     sput-boolean v3, Lcom/samsung/android/smartsuggestions/service/smartreply/SmartReplyResolver;->cacheSmartReplyAvailable:Z
 
-    invoke-static {v1}, Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;->getInstance(Landroid/content/Context;)Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;
-
-    move-result-object v0
-
-    invoke-virtual {v0}, Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;->getPolicyListSync()Lcom/samsung/android/smartsuggestions/service/policy/Policylist;
-
-    move-result-object v0
-
-    if-eqz v0, :cond_rezoss_force_policy_reply_done
-
-    invoke-virtual {v0}, Lcom/samsung/android/smartsuggestions/service/policy/Policylist;->getSuggestedReplyPackageAllowlist()Ljava/util/List;
-
-    move-result-object v0
-
-    invoke-interface {v0, v2}, Ljava/util/List;->contains(Ljava/lang/Object;)Z
-
-    move-result v0
-
-    if-nez v0, :cond_rezoss_force_chat_reply_available
-
-    :cond_rezoss_force_policy_reply_done
-    const-string v0, "com.whatsapp"
-
-    invoke-static {v2, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v0
-
-    if-nez v0, :cond_rezoss_force_chat_reply_available
-
-    const-string v0, "org.telegram.messenger"
-
-    invoke-static {v2, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v0
-
-    if-nez v0, :cond_rezoss_force_chat_reply_available
-
-    const-string v0, "org.telegram.messenger.web"
-
-    invoke-static {v2, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v0
-
-    if-nez v0, :cond_rezoss_force_chat_reply_available
-
-    const-string v0, "com.whatsapp.w4b"
-
-    invoke-static {v2, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v0
-
-    if-nez v0, :cond_rezoss_force_chat_reply_available
-
-    const-string v0, "com.samsung.android.messaging"
-
-    invoke-static {v2, v0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v0
-
-    if-eqz v0, :cond_rezoss_force_chat_reply_done
-
-    :cond_rezoss_force_chat_reply_available
-    const/4 v3, 0x1
-
-    sput-boolean v3, Lcom/samsung/android/smartsuggestions/service/smartreply/SmartReplyResolver;->cacheSmartReplyAvailable:Z
-
-    const-string v0, "SmartReplyResolver[reply]"
-
-    const-string/jumbo v9, "force-enabled chat SmartReply availability"
-
-    invoke-static {v0, v9}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    :cond_rezoss_force_chat_reply_done
     if-nez v3, :cond_1a
 
     if-nez v8, :cond_1a
@@ -2472,77 +2399,6 @@
     goto/16 :goto_10
 
     :goto_13
-    invoke-static {v1}, Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;->getInstance(Landroid/content/Context;)Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;
-
-    move-result-object v3
-
-    invoke-virtual {v3}, Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;->getPolicyListSync()Lcom/samsung/android/smartsuggestions/service/policy/Policylist;
-
-    move-result-object v3
-
-    if-eqz v3, :cond_rezoss_post_recall_policy_done
-
-    invoke-virtual {v3}, Lcom/samsung/android/smartsuggestions/service/policy/Policylist;->getSuggestedReplyPackageAllowlist()Ljava/util/List;
-
-    move-result-object v3
-
-    invoke-interface {v3, v2}, Ljava/util/List;->contains(Ljava/lang/Object;)Z
-
-    move-result v3
-
-    if-nez v3, :cond_rezoss_force_post_recall_reply
-
-    :cond_rezoss_post_recall_policy_done
-    const-string v3, "com.whatsapp"
-
-    invoke-static {v2, v3}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v3
-
-    if-nez v3, :cond_rezoss_force_post_recall_reply
-
-    const-string v3, "org.telegram.messenger"
-
-    invoke-static {v2, v3}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v3
-
-    if-nez v3, :cond_rezoss_force_post_recall_reply
-
-    const-string v3, "org.telegram.messenger.web"
-
-    invoke-static {v2, v3}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v3
-
-    if-nez v3, :cond_rezoss_force_post_recall_reply
-
-    const-string v3, "com.whatsapp.w4b"
-
-    invoke-static {v2, v3}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v3
-
-    if-nez v3, :cond_rezoss_force_post_recall_reply
-
-    const-string v3, "com.samsung.android.messaging"
-
-    invoke-static {v2, v3}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v3
-
-    if-eqz v3, :cond_rezoss_post_recall_reply_done
-
-    :cond_rezoss_force_post_recall_reply
-    const/16 v17, 0x1
-
-    const-string v3, "SmartReplyResolver[reply]"
-
-    const-string/jumbo v4, "force chat reply execution after recall"
-
-    invoke-static {v3, v4}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    :cond_rezoss_post_recall_reply_done
     if-eqz v17, :cond_24
 
     sget-object v0, Lcom/samsung/android/smartsuggestions/service/translation/session/KeyboardChecker;->INSTANCE:Lcom/samsung/android/smartsuggestions/service/translation/session/KeyboardChecker;
@@ -2551,77 +2407,6 @@
 
     move-result v0
 
-    invoke-static {v1}, Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;->getInstance(Landroid/content/Context;)Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;
-
-    move-result-object v3
-
-    invoke-virtual {v3}, Lcom/samsung/android/smartsuggestions/service/policy/PolicyManager;->getPolicyListSync()Lcom/samsung/android/smartsuggestions/service/policy/Policylist;
-
-    move-result-object v3
-
-    if-eqz v3, :cond_rezoss_keyboard_policy_done
-
-    invoke-virtual {v3}, Lcom/samsung/android/smartsuggestions/service/policy/Policylist;->getSuggestedReplyPackageAllowlist()Ljava/util/List;
-
-    move-result-object v3
-
-    invoke-interface {v3, v2}, Ljava/util/List;->contains(Ljava/lang/Object;)Z
-
-    move-result v3
-
-    if-nez v3, :cond_rezoss_force_chat_keyboard_visible
-
-    :cond_rezoss_keyboard_policy_done
-    const-string v3, "com.whatsapp"
-
-    invoke-static {v2, v3}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v3
-
-    if-nez v3, :cond_rezoss_force_chat_keyboard_visible
-
-    const-string v3, "org.telegram.messenger"
-
-    invoke-static {v2, v3}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v3
-
-    if-nez v3, :cond_rezoss_force_chat_keyboard_visible
-
-    const-string v3, "org.telegram.messenger.web"
-
-    invoke-static {v2, v3}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v3
-
-    if-nez v3, :cond_rezoss_force_chat_keyboard_visible
-
-    const-string v3, "com.whatsapp.w4b"
-
-    invoke-static {v2, v3}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v3
-
-    if-nez v3, :cond_rezoss_force_chat_keyboard_visible
-
-    const-string v3, "com.samsung.android.messaging"
-
-    invoke-static {v2, v3}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result v3
-
-    if-eqz v3, :cond_rezoss_chat_keyboard_visible_done
-
-    :cond_rezoss_force_chat_keyboard_visible
-    const/4 v0, 0x1
-
-    const-string v3, "SmartReplyResolver[reply]"
-
-    const-string/jumbo v4, "force chat keyboard-visible gate for suggested replies"
-
-    invoke-static {v3, v4}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    :cond_rezoss_chat_keyboard_visible_done
     if-eqz v0, :cond_23
 
     move-object/from16 v0, p0
@@ -3876,65 +3661,23 @@
 
     move-result-object p0
 
-    if-eqz p0, :cond_rezoss_recall_policy_done
+    if-eqz p0, :cond_rezoss_recall_original
 
     invoke-virtual {p0}, Lcom/samsung/android/smartsuggestions/service/policy/Policylist;->getSuggestedReplyPackageAllowlist()Ljava/util/List;
 
     move-result-object p0
 
-    invoke-interface {p0, p2}, Ljava/util/List;->contains(Ljava/lang/Object;)Z
-
-    move-result p0
-
-    if-eqz p0, :cond_rezoss_recall_policy_done
-
-    const-string p0, "SmartReplyResolver[reply]"
-
-    const-string/jumbo p1, "allow policy chat recall/action despite policy blocklist"
-
-    invoke-static {p0, p1}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    const/4 p0, 0x1
-
-    return p0
-
-    :cond_rezoss_recall_policy_done
-    const-string p0, "com.whatsapp"
-
-    invoke-static {p2, p0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result p0
-
-    if-nez p0, :cond_rezoss_recall_allow_whatsapp
-
-    const-string p0, "org.telegram.messenger"
-
-    invoke-static {p2, p0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result p0
-
-    if-nez p0, :cond_rezoss_recall_allow_whatsapp
-
-    const-string p0, "org.telegram.messenger.web"
-
-    invoke-static {p2, p0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
-
-    move-result p0
-
-    if-nez p0, :cond_rezoss_recall_allow_whatsapp
-
-    const-string p0, "com.whatsapp.w4b"
+    if-eqz p0, :cond_rezoss_recall_original
 
-    invoke-static {p2, p0}, Lkotlin/jvm/internal/h;->j(Ljava/lang/Object;Ljava/lang/Object;)Z
+    invoke-interface {p0, p2}, Ljava/util/List;->contains(Ljava/lang/Object;)Z
 
     move-result p0
 
     if-eqz p0, :cond_rezoss_recall_original
 
-    :cond_rezoss_recall_allow_whatsapp
     const-string p0, "SmartReplyResolver[reply]"
 
-    const-string/jumbo p1, "allow chat recall/action despite policy blocklist"
+    const-string/jumbo p1, "allow suggested-reply package recall/action"
 
     invoke-static {p0, p1}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
 
@@ -3943,6 +3686,7 @@
     return p0
 
     :cond_rezoss_recall_original
+
     sget-object p0, Lcom/samsung/android/smartsuggestions/service/smartreply/SmartReplyResolver;->recallActionAllowedList:Ljava/util/concurrent/atomic/AtomicReference;
 
     invoke-virtual {p0}, Ljava/util/concurrent/atomic/AtomicReference;->get()Ljava/lang/Object;
